### DISCUSSION TITLE
Filter for entity with array that contains element?

### DISCUSSION THREAD

**Discussion by Atlinx** [twinlogix/typetta#271] [Q&A] [answered]
How can I filter for an entity whose array contains an element? I have an entity like so:
```graphql
  type Config @entity @mongodb {
    id: ID! @id(from: "db") @alias(value: "_id")
    
    textChannelIds: [String!]
    voiceChannelIds: [String!]
    categoryId: String
  }
```

I'd like to filter for all version of the `Config` model that has a specific id (E.x. "2093470923843") inside it's textChannelIds field.
Note that `textChannelIds`, `voiceChannelIds`, and `categoryId` have nothing to do with the models in a schema -- they are merely data.
I'm using GraphQL to make this query, so I can't use native MongoDB queries without making a dedicated resolver.

I've tried `contains`, but then I found out that was a `string` only operation.

**Comment by edobrb** [comment, marked as answer]
At the moment it's not supported. We'll work on it ASAP.

A possible workaround could be to transform your model as follow:
```graphql
type Config @entity @mongodb {
    id: ID! @id(from: "db") @alias(value: "_id")
    
    textChannelIds: [ConfigTextChannel!] @foreignRef
    voiceChannelIds: [ConfigVoiceChannel!] @foreignRef
    categoryId: String
}

type ConfigTextChannel @entity @mongodb {
    id: ID! @id(from: "db") @alias(value: "_id")
    configId: ID!
    value: String!
}

type ConfigVoiceChannel @entity @mongodb {
    id: ID! @id(from: "db") @alias(value: "_id")
    configId: ID!
    value: String!
}
```

And executing the find on the `ConfigTextChannel` or the `ConfigVoiceChannel` and then find the `Config`

**Comment by edobrb** [comment]
@Atlinx if you want try out the prerelease [2.1.4-1](https://www.npmjs.com/package/@twinlogix/typetta/v/2.1.4-1) it has the feature that you suggested. The GraphQL query become:
```graphql
query {
  configs(filter: { textChannelIds: { has: "2093470923843" } }) {
    id
  }
}
```
